preprocessing/structs.ts

```diff
@@ -96,121 +161,159 @@ const id_field = (name: string) => {
+
+      We store the density as a float with 2 bits for the mantissa, 7 for the exponent, and an exponent bias of 32, so that density ranges from $2^{-32}$ to $1.75 \times 2^{95}$. For more information, see `DensityLib`.
+      
+      */
+      { name: "density", bits: 9, type: "Density", underlyingType: "uint"},
+      /* To save on gas, Mangrove cache the entire tick tree branch of the bin that contains the best offer in each offer list's `local` parameter. Taken together, `binPosInLeaf`, `level3`, `level2`, `level1` and `root` provide the following info:
+      - What the current bin is (see `BinLib.bestBinFromLocal`)
+      - When a leaf is emptied and the next offer must be fetched, the information in the fields `level3`, `level2`, `level1` and `root` avoid multiple storage reads
+      */
+      { name: "binPosInLeaf", bits: 2, type: "uint" },
+      { name: "level3", bits: 64, type: "Field", underlyingType: "uint" },
+      { name: "level2", bits: 64, type: "Field", underlyingType: "uint" },
+      { name: "level1", bits: 64, type: "Field", underlyingType: "uint" },
+      { name: "root", bits: 2, type: "Field", underlyingType: "uint" },
+      /* * `offer_gasbase` represents the gas overhead used by processing the offer inside Mangrove + the overhead of initiating an entire order. Mangrove considers that a failed offer has used at least `offer_gasbase` gas. The actual field name is `kilo_offer_gasbase` and the accessor `offer_gasbase` returns `kilo_offer_gasbase*1e3`. Local to an offer list, because the costs of calling `outbound_tkn` and `inbound_tkn`'s `transferFrom` are part of `offer_gasbase`. Should only be updated when ERC20 contracts change or when opcode prices change. */
+      fields.kilo_offer_gasbase,
+      /* * If `lock` is true, orders may not be added nor executed, nor the offer list read by external contracts.
+
+        Reentrancy during offer execution is not considered safe:
+      * during execution, an offer could consume other offers further up in the list, effectively frontrunning the taker currently executing the offer.
+      * it could also cancel other offers, creating a discrepancy between the advertised and actual market price at no cost to the maker.
+      * an offer insertion consumes an unbounded amount of gas (because it has to be correctly placed in the book).
+
+  Note: An optimization in the `marketOrder` function relies on reentrancy being forbidden.
+      */
+      { name: "lock", bits: 1, type: "bool" },
+      /* * `best` holds the current best offer id. Has size of an id field. *Danger*: reading best inside a lock may give you a stale value. */
+      // id_field("best"),
```

Comment:
Formatting is off in the generated html:

![billede](https://github.com/mangrovedao/mangrove-core/assets/3458856/9e8d73f2-352a-4f34-8fd9-c5c71c8e6786)

Reply:
the entire thing should be removed